### DISCUSSION TITLE
Timeframe of Data

### DISCUSSION THREAD

**Discussion by drussman** [socius-org/RedditHarbor#13] [Q&A] [answered]
This is a great resource for researchers, thanks Nick and co.! How far back, in terms of the timeframe, can RedditHarbor collect?

**Comment by nick-sh-oh** [comment, marked as answer]
Hi Daniel, thank you for your question!

That's an excellent point you've raised, and it's something I've personally explored during RedditHarbor's beta testing phase around July 2023. At that time, I was collecting submissions from one of the [subreddit collections](https://github.com/socius-org/RedditHarbor/blob/main/redditharbor/utils/subreddit_collections.py) over a 24-hour period. While RedditHarbor demonstrated the capability to collect historical data as far back as 2009, it appeared to perform best when collecting data from the past 5-6 months, as shown in the plot below.

<img width="573" alt="image" src="https://github.com/socius-org/RedditHarbor/assets/130935698/8a63a17a-50e5-4dbb-b1c7-28b97a5ddde7">

This is largely due to Reddit Data API, as unlike Twitter Data API, they do not support historical data search. Reddit Data API's strength lies in efficiently collecting and processing more recent submissions. 

If you have any further questions or need additional clarification, please feel free to ask.
